### PR TITLE
Raise Exception from AnyInstanceOf Constructor

### PR DESCRIPTION
**Pull request by EricLiclair**
**What:**

#233 

**Why:**

**How:**

Checked for the input arguments of the `AnyInstanceOf` to be of an instance of `type`. If not, raise a `ValueError` exception.

**Risks:**

**Checklist**:

- [x] Added tests, if you've added code that should be tested
- [x] Updated the documentation, if you've changed APIs
- [x] Ensured the test suite passes
- [x] Made sure your code lints
- [x] Completed the Contributor License Agreement ("CLA")

Do I need to add the "correct usage" of the AnyInstanceOf in the tests?

### CODE REVIEW

**Review by macisamuele — APPROVED** [review]
@EricLiclair thanks for contributing

> Do I need to add the "correct usage" of the AnyInstanceOf in the tests?

`testAnyInstanceOf` does already check for it ;)

File: tests/sample_module.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class SomeClass:
     other_class_attribute = OtherClass
     typedattr: str = "bruh"
 
-    def method(self):
+    def method(self, *args, **kwargs):
```
Comment (EricLiclair) [inline tests/sample_module.py:22]: This was used for writing a different test case which I removed before committing. The removed test was to test the "correct usage" of the `AnyInstanceOf `using `mock_callable`. If asked to, I'll add that test again.

**Comment by EricLiclair** [comment]
Thank you @macisamuele for your help. Learned a lot in the past week. 😊

File: tests/sample_module.py (hunk 1 above)
Comment (macisamuele) [inline tests/sample_module.py:22]: If these changes are no longer needed then they can be removed

File: tests/matchers_unittest.py [hunk 2]
```diff
@@ -133,6 +133,14 @@ def testAnyInstanceOf(self):
         self.assertEqual(testslide.matchers.AnyInstanceOf(str), "durrdurr")
         self.assertNotEqual(testslide.matchers.AnyInstanceOf(str), 7)
 
+    def testTriggeringAnyInstanceOfException(self):
+        c = sample_module.SomeClass()
+        with self.assertRaises(ValueError):
+            self.mock_callable(c, "method").for_call(
+                testslide.matchers.AnyInstanceOf(2)
+            ).to_return_value("mocked_response")
+            c.method()
```
Comment (macisamuele) [inline tests/matchers_unittest.py:142]: `testslide.matchers.AnyInstanceOf(2)` would raise the exception so invoking `c.method()` is not needed as well as creating the `c` instance is not neede.

Yu might want to have this test more inline with the other tests

```suggestion
        self.assertNotEqual(testslide.matchers.AnyInstanceOf(str), 7)
        with self.assertRaises(ValueError):
            testslide.matchers.AnyInstanceOf(2)
```

File: tests/sample_module.py (hunk 1 above)
Comment (EricLiclair) [reply tests/sample_module.py:22]: Shall I remove it and commit again?

File: tests/matchers_unittest.py [hunk 3]
```diff
@@ -131,7 +131,11 @@ def testAny(self):
 
     def testAnyInstanceOf(self):
         self.assertEqual(testslide.matchers.AnyInstanceOf(str), "durrdurr")
+
+    def testTriggeringAnyInstanceOfException(self):
```
Comment (macisamuele) [inline tests/matchers_unittest.py:135]: nit. I would suggest to keep a single test just to keep it uniform with the rest of the tests

**Review by macisamuele — COMMENTED** [review]
I know that I might asking a lot, so sorry in advance.
Would you have bandwidth to address also the other matches? (possibility on a different PR such that this can eventually be merged)

File: tests/matchers_unittest.py (hunk 3 above)
Comment (EricLiclair) [reply tests/matchers_unittest.py:135]: Okay, I believe the `self.assertNotEqual(testslide.matchers.AnyInstanceOf(str), 7)` was unintentionally selected to be added to the triggering test.

I'll keep it where it was initially, in `testAnyInstanceOf`.
And only add a single triggerTest to `testTriggeringAnyInstanceOfException`.

Is it what I am expected to do?

**Comment by EricLiclair** [comment]
> I know that I might asking a lot, so sorry in advance.
> Would you have bandwidth to address also the other matches? (possibility on a different PR such that this can eventually be merged)

And yes, no worries of the bandwidth.
Also, I might sometimes ask childish questions 😅. Just to be sure about what needs to be done. I hope it won't be an issue.

**Comment by macisamuele** [comment]
> And yes, no worries of the bandwidth.
> Also, I might sometimes ask childish questions 😅. Just to be sure about what needs to be done. I hope it won't be an issue.

at this point I do believe that you have fully grasped what needs to be done and how to do it. Still, no questions are childish so do not hesitate to post them.

**Pull request closed without merging**